internal/service/advancedcluster/model_advanced_cluster_test.go

```diff
@@ -21,6 +23,128 @@ var (
 	advancedClusters = []admin.AdvancedClusterDescription{{StateName: conversion.StringPtr("NOT IDLE")}}
 )
 
+func TestFlattenReplicationSpecs(t *testing.T) {
+	var (
+		regionName         = "EU_WEST_1"
+		providerName       = "AWS"
+		expectedID         = "id1"
+		unexpectedID       = "id2"
+		expectedZoneName   = "z1"
+		unexpectedZoneName = "z2"
+		regionConfigAdmin  = []admin.CloudRegionConfig{{
+			ProviderName: &providerName,
+			RegionName:   &regionName,
+		}}
+		regionConfigTf = map[string]any{
+			"provider_name": "AWS",
+			"region_name":   regionName,
+		}
+		regionConfigTfDiffZone = map[string]any{
+			"provider_name": "AWS",
+			"region_name":   regionName,
+			"zone_name":     unexpectedZoneName,
+		}
+		admin1     = admin.ReplicationSpec{Id: &expectedID, ZoneName: &expectedZoneName, RegionConfigs: &regionConfigAdmin}
+		admin2     = admin.ReplicationSpec{Id: &unexpectedID, ZoneName: &unexpectedZoneName, RegionConfigs: &regionConfigAdmin}
+		testSchema = map[string]*schema.Schema{
+			"project_id": {Type: schema.TypeString},
+		}
+		tf1SameIDSameZone = map[string]any{
+			"id":             expectedID,
+			"num_shards":     1,
+			"region_configs": []any{regionConfigTf},
```

Comment:
knit: it seems that regionConfigTf, regionConfigTfDiffZone are only used for elements in []any, maybe we can just have them from that type so we just do: 
"region_configs": regionConfigTf (maybe doing the var name plural like regionConfigsTf)

Reply:
Done in 9425ff49b8b3e0c3b0dad38fc44160d40e88033b

Reply:
Reverted in 838340f26e0eb46aca8a8a9cadadc31cd1ae07f2
See [error in test](https://github.com/mongodb/terraform-provider-mongodbatlas/actions/runs/9580418592/job/26415016399)